components/greaselion/browser/greaselion_service_impl.cc

```diff
@@ -178,16 +179,13 @@ scoped_refptr<Extension> ConvertGreaselionRuleToExtensionOnTaskRunner(
   if (!extension.get()) {
     LOG(ERROR) << "Could not load Greaselion extension";
     LOG(ERROR) << error;
-    return nullptr;
+    return base::nullopt;
   }
 
   // Take ownership of this temporary directory so it's deleted when
   // the service exits
-  if (extension_dirs) {
-    extension_dirs->push_back(std::move(temp_dir));
-  }
-
-  return extension;
+  return base::Optional<greaselion::GreaselionConvertedExtension>(
```

Comment:
afaik it is possible to return `std::make_pair(...)` so `Optional` would be implicitly constructed

Reply:
OK, fixed.